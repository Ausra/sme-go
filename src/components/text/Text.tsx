import { FunctionComponent } from "react";
import styled from "styled-components/macro";
import { defaultTheme, typeScale } from "../../utils/global-styles";

type TextSize = "small" | "medium" | "large";

interface TextProps {
  dataTestId?: string;
  children?: string | number;
  size?: TextSize;
  color?: string;
}

const defaultTestId = "styled-text";

const StyledText = styled.p<{ size?: TextSize; color?: string }>`
  font-size: ${(props) =>
    props.size ? typeScale[props.size] : typeScale.medium};
  color: ${(props) => (props.color ? props.color : defaultTheme.textColor)};
  pointer-events: none;
`;

const Text: FunctionComponent<TextProps> = ({
  dataTestId,
  children,
  size,
  color,
}) => {
  return (
    <StyledText
      size={size}
      data-testid={dataTestId || defaultTestId}
      color={color}
    >
      {children}
    </StyledText>
  );
};

export default Text;
