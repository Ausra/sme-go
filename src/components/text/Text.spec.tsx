import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import Text from "./Text";
import "jest-styled-components";

afterEach(cleanup);

it("should render default Text", () => {
  render(<Text />);
  const text = screen.getByTestId("styled-text");
  expect(text).toMatchSnapshot();
});
