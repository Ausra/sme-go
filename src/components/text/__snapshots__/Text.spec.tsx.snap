// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default Text 1`] = `
.c0 {
  font-size: 1rem;
  color: #000000;
}

<p
  class="c0"
  data-testid="styled-text"
/>
`;
